vital/plugin_loader/plugin_registrar.h

```diff
@@ -0,0 +1,86 @@
+/*ckwg +29
+ * Copyright 2018 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PLUGIN_LOADER_PLUGIN_REGISTRAR_H
+#define PLUGIN_LOADER_PLUGIN_REGISTRAR_H
+
+#include <vital/plugin_loader/plugin_factory.h>
+
+// ==================================================================
+// Support for adding factories for the plugin loader
+
+namespace kwiver {
+namespace vital {
+  class plugin_loader;
+} // end namespace vitak
+
+
+/// Class to assist in registering tools.
+class plugin_registrar
+{
+public:
+  /**
+   * @brief Create registrar
+   *
+   * This class contains the common data used for registering tools.
+   *
+   * @param vpl Reference to the plugin loader
+   * @param name Name of this loadable module.
+   */
+  plugin_registrar( vital::plugin_loader& vpl,
+                    const std::string& name )
+    : mod_name( name )
+
+#if defined PLUGIN_ORG
```

Comment:
Does this require you include "plugin_organization.h" before this file to work as expected?  I'm not a fan of anything that requires you to include files in a specific order.  Could this just be a third optional argument to the constructor that defaults to "Unspecified"?  Then the caller can pass in `PLUGIN_ORG` if they want.  Or it could be a setter member function if that is easier.